apps/trading/client-pages/markets/closed.tsx

```diff
@@ -89,12 +92,25 @@ export const Closed = () => {
       bestOfferPrice: market.data?.bestOfferPrice,
       markPrice: market.data?.markPrice,
       settlementDataOracleId:
-        instrument.product.dataSourceSpecForSettlementData.id,
+        'dataSourceSpecForSettlementData' in instrument.product
+          ? instrument.product.dataSourceSpecForSettlementData.id
+          : '',
       settlementDataSpecBinding,
-      setlementDataSourceFilter: filter,
+      settlementDataSourceFilter: filter,
       tradingTerminationOracleId:
-        instrument.product.dataSourceSpecForTradingTermination.id,
-      settlementAsset: instrument.product.settlementAsset,
+        'dataSourceSpecForTradingTermination' in instrument.product
+          ? instrument.product.dataSourceSpecForTradingTermination.id
+          : '',
+      settlementAsset:
+        'settlementAsset' in instrument.product
```

Comment:
getAsset function? I'd avoid having some default values here, it could result in errors further down the component tree